contrib/elastic/go-elasticsearch.v5/elastictrace_v7_test.go

```diff
@@ -0,0 +1,243 @@
+	assert.NoError(err)
+
+	_, err = esapi7.GetRequest{
+		Index:      "not-real-index",
+		DocumentID: "1",
+	}.Do(context.TODO(), client)
+	assert.NoError(err)
+
+	span := mt.FinishedSpans()[0]
+	assert.Equal(`{"error":{`, span.Tag(ext.Error).(error).Error())
+}
+
+func TestClientV7Failure(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	cfg := elasticsearch7.Config{
+		Transport: NewRoundTripper(WithServiceName("my-es-service")),
+		Addresses: []string{
+			"http://127.0.0.1:9207", // inexistent service, it must fail
+		},
+	}
+	client, err := elasticsearch7.NewClient(cfg)
+	assert.NoError(err)
+
+	_, err = esapi7.IndexRequest{
+		Index:        "twitter",
+		DocumentID:   "1",
+		DocumentType: "tweet",
+		Body:         strings.NewReader(`{"user": "test", "message": "hello"}`),
+	}.Do(context.TODO(), client)
+	assert.Error(err)
+
+	spans := mt.FinishedSpans()
+	checkPUTTrace(assert, mt)
+
+	assert.NotEmpty(spans[0].Tag(ext.Error))
+	assert.Equal("*net.OpError", fmt.Sprintf("%T", spans[0].Tag(ext.Error).(error)))
+}
+
+func TestResourceNamerSettingsV7(t *testing.T) {
+	staticName := "static resource name"
+	staticNamer := func(url, method string) string {
+		return staticName
+	}
+
+	t.Run("default", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		cfg := elasticsearch7.Config{
+			Transport: NewRoundTripper(),
+			Addresses: []string{
+				elasticV7URL,
+			},
+		}
+		client, err := elasticsearch7.NewClient(cfg)
+		assert.NoError(t, err)
+
+		_, err = esapi7.GetRequest{
+			Index:        "logs_2017_05/event/_search",
+			DocumentID:   "1",
+			DocumentType: "tweet",
+		}.Do(context.TODO(), client)
+
+		span := mt.FinishedSpans()[0]
+		assert.Equal(t, "GET /logs_?_?/event/_search/tweet/?", span.Tag(ext.ResourceName))
+	})
+
+	t.Run("custom", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		cfg := elasticsearch7.Config{
+			Transport: NewRoundTripper(WithResourceNamer(staticNamer)),
+			Addresses: []string{
+				elasticV7URL,
+			},
+		}
+		client, err := elasticsearch7.NewClient(cfg)
+		assert.NoError(t, err)
+
+		_, err = esapi7.GetRequest{
+			Index:        "logs_2017_05/event/_search",
+			DocumentID:   "1",
+			DocumentType: "tweet",
+		}.Do(context.TODO(), client)
+
+		span := mt.FinishedSpans()[0]
+		assert.Equal(t, staticName, span.Tag(ext.ResourceName))
+	})
+}
+
+func TestAnalyticsSettingsV7(t *testing.T) {
+	assertRate := func(t *testing.T, mt mocktracer.Tracer, rate interface{}, opts ...ClientOption) {
+
+		cfg := elasticsearch7.Config{
+			Transport: NewRoundTripper(opts...),
+			Addresses: []string{
+				elasticV7URL,
+			},
+		}
+		client, err := elasticsearch7.NewClient(cfg)
+		assert.NoError(t, err)
+
+		_, err = esapi7.IndexRequest{
+			Index:        "twitter",
+			DocumentID:   "1",
+			DocumentType: "tweet",
+			Body:         strings.NewReader(`{"user": "test", "message": "hello"}`),
+		}.Do(context.TODO(), client)
```

Comment:
`context.Background()` is fine for this case. We don't want to indicate this is not complete.